plugins/discourse-subscriptions/test/javascripts/integration/components/campaign-banner-test.gjs

```diff
@@ -0,0 +1,142 @@
+import { render } from "@ember/test-helpers";
+import { module, test } from "qunit";
+import sinon from "sinon";
+import { setupRenderingTest } from "discourse/tests/helpers/component-test";
+import pretender from "discourse/tests/helpers/create-pretender";
+import CampaignBanner from "discourse/plugins/discourse-subscriptions/discourse/components/campaign-banner";
+
+module("Subscriptions | campaign-banner", function (hooks) {
+  setupRenderingTest(hooks);
+
+  test("does not make AJAX request when plugin is disabled", async function (assert) {
+    // Set up pretender to track requests
+    let requestCount = 0;
+    pretender.get("/s/contributors", () => {
+      requestCount++;
+      return [200, { "Content-Type": "application/json" }, "[]"];
+    });
+
+    // Set site settings
+    this.owner.lookup("service:site-settings").discourse_subscriptions_enabled =
+      false;
+    this.owner.lookup(
+      "service:site-settings"
+    ).discourse_subscriptions_campaign_enabled = true;
+    this.owner.lookup(
+      "service:site-settings"
+    ).discourse_subscriptions_campaign_show_contributors = true;
+
+    // Mock current user
+    this.owner.lookup("service:current-user").setProperties({
+      id: 1,
+      username: "testuser",
+    });
+
+    // Mock site settings for show_campaign_banner
+    this.owner.lookup("service:site").setProperties({
+      show_campaign_banner: true,
+    });
+
+    // Mock router with a valid route using sinon stub
+    sinon
+      .stub(this.owner.lookup("service:router"), "currentRouteName")
+      .value("discovery.latest");
+
+    await render(<template><CampaignBanner /></template>);
+
+    assert.strictEqual(
+      requestCount,
+      0,
+      "No AJAX request is made when plugin is disabled"
+    );
+  });
+
+  test("renders banner and makes AJAX request when all conditions are met", async function (assert) {
```

Comment:
Given all the pretender/mocking here, this may be easier to write/read as a system spec.